pubsub.go

```diff
@@ -31,6 +31,10 @@ var (
 	// Use WithSeenMessagesTTL to configure this per pubsub instance, instead of overriding the global default.
 	TimeCacheDuration = 120 * time.Second
 
+	// TimeCacheStrategy specifies which type of lookup/cleanup strategy is used by the seen messages cache.
+	// Use WithSeenMessagesStrategy to configure this per pubsub instance, instead of overriding the global default.
+	TimeCacheStrategy = timecache.Strategy_LastSeen
```

Comment:
A bit confusing, the old default is still present in 
https://github.com/libp2p/go-libp2p-pubsub/blob/04bfcf58514f59bddb650fec5b2edac018c8c406/timecache/time_cache.go#L19
and 
https://github.com/libp2p/go-libp2p-pubsub/blob/04bfcf58514f59bddb650fec5b2edac018c8c406/timecache/time_cache.go#L28-L30

Reply:
Good point! Fixed.

Reply:
I would prefer to keep the existing behaviour as default.

I dont like surprising users with subtle changes, esp since this might effect memory usage.

Note that kubo is a fringe user, not surprising filecoin and eth is much more important here.